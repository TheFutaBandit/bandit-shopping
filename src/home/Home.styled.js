import {styled} from 'styled-components';
import homebodypicture from '../assets/homebodypicture.svg'

export const Header = styled.div`
    height: 82.567px;
    color: red;
`

export const Container = styled.div`
    display: flex;
    flex-direction: column;
    align-items: center;
    font-family: 'Roboto Condensed', sans-serif;
`

export const Mainbody_styled = styled.div`
    display: flex;
    justify-content: center;
    margin-top: 1rem;   
    margin-bottom: 1rem;
`

export const Mainbodypicture = styled.img.attrs({
    src: homebodypicture
})`
    max-width: 850px;
    height: auto;
`

export const Footer_styled = styled.div`
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
    margin-top: 1rem;
    position: relative;
    gap: 20px;
`

export const FooterText = styled.p`
    box-sizing: border-box;
    margin:0px;
    font-size: ${prop => prop.size};
    margin: ${prop => prop.padding};
    align-self: ${prop => prop.alignment};
    font-weight: 900;
`

FooterText.defaultProps = {
    size : "1rem",
    padding: "0px 0px 0px 0px",
    alignment: "flex-start",
}

/* top */

// position: absolute;
// width: 137px;
// height: 26px;
// left: calc(50% - 137px/2 + 0.5px);
// top: 158px;

