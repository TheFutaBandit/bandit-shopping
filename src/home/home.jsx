import { Mainbody_styled, Mainbodypicture, Footer_styled, FooterText } from "./Home.styled";

const MainBody = () => {
    return (
        <Mainbody_styled>
            <Mainbodypicture />
        </Mainbody_styled>
    )
}

export const Footer = () => {
    return (
        <Footer_styled>
            <FooterText size="2rem" padding="0px 0px 0px 0px" alignment="center">SHOP AT THE BANDIT'S STASH</FooterText>
            <FooterText size="1rem" padding="0px 0px 0px -100px" alignment="flex-start">SHIT SO CHEAP WE MAKE IT STOLEN</FooterText>
            <FooterText size="1rem" padding="0px -50px 0px 0px" alignment="flex-end">SHIT SO BAD ITS A BANDITâ€™S DOING</FooterText>
            <FooterText size="1rem" padding="0px 0px 0px -20px" alignment="center">THE BANDIT - KINGS OF LEON.</FooterText>
        </Footer_styled>
    )
}

export const Home = () => {

    return (
        <>
            <MainBody />
            <Footer />
        </>
    )
}