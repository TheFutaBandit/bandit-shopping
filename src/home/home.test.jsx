import {describe, it, expect} from 'vitest';
import { screen, render} from '@testing-library/react'
import {Home} from './home.jsx'

describe("check home rendering", () => {
    it("home render", () => {
        const { container } = render(<Home />)
        expect(container).toMatchSnapshot();
    })
})