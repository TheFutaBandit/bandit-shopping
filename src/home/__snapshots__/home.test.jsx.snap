// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`check home rendering > home render 1`] = `
<div>
  <div
    class="sc-dVBluf eOhEwE"
  >
    <img
      class="sc-fLDLck htvQgj"
      src="/src/assets/homebodypicture.svg"
    />
  </div>
  <div
    class="sc-dycHyt ecLbcZ"
  >
    <p
      alignment="center"
      class="sc-jbAkgO jbgKnx"
      padding="0px 0px 0px 0px"
    >
      SHOP AT THE BANDIT'S STASH
    </p>
    <p
      alignment="flex-start"
      class="sc-jbAkgO eLSLAn"
      padding="0px 0px 0px -100px"
    >
      SHIT SO CHEAP WE MAKE IT STOLEN
    </p>
    <p
      alignment="flex-end"
      class="sc-jbAkgO fkEWjQ"
      padding="0px -50px 0px 0px"
    >
      SHIT SO BAD ITS A BANDITâ€™S DOING
    </p>
    <p
      alignment="center"
      class="sc-jbAkgO kvZRuN"
      padding="0px 0px 0px -20px"
    >
      THE BANDIT - KINGS OF LEON.
    </p>
  </div>
</div>
`;
